wp-env: Add support for different options in each environment (#22568)
WordPress/gutenberg: packages/env/README.md

`.wp-env.json` supports six fields for options applicable to both the tests and development instances.

| Field        | Type           | Default                                | Description                                                                                                               |
| ------------ | -------------- | -------------------------------------- | ------------------------------------------------------------------------------------------------------------------------- |
| `"core"`     | `string\|null` | `null`                                 | The WordPress installation to use. If `null` is specified, `wp-env` will use the latest production release of WordPress.  |
| `"plugins"`  | `string[]`     | `[]`                                   | A list of plugins to install and activate in the environment.                                                             |
| `"themes"`   | `string[]`     | `[]`                                   | A list of themes to install in the environment. The first theme in the list will be activated.                            |

> | `"port"`     | `integer`      | `8888` (`8889` for the tests instance) | The primary port number to use for the insallation. You'll access the instance through the port: 'http://localhost:8888'. |

Reviewer: ```suggestion
| `"port"`     | `integer`      | `8888` (`8889` for the tests instance) | The primary port number to use for the installation. You'll access the instance through the port: 'http://localhost:8888'. |
```

Remote sources will be downloaded into a temporary directory located in `~/.wp-env`.

> Additionally, the key `env` is available to override any of the above options on an individual-environment basis. For example, take following `.wp-env.json` file:

Reviewer: ```suggestion
Additionally, the key `env` is available to override any of the above options on an individual-environment basis. For example, take the following `.wp-env.json` file:
```

```json
{
	"plugins": [ "." ],
	"config": {
		"KEY_1": true,
		"KEY_2": false
	},
	"env": {
		"development": {
			"themes": [ "./one-theme" ]
		},
		"tests": {
			"config": {
				"KEY_1": false
			},
			"port": 3000
		}
	}
}
```

On the development instance, `cwd` will be mapped as a plugin, `one-theme` will be mapped as a theme, KEY_1 will be set to true, and KEY_2 will be set to false. Also note that the default port, 8888, will be used as well.

On the tests instance, `cwd` is still mapped as a plugin, but no theme is mapped. Additionaly, while KEY_2 is still set to false, KEY_1 is overriden and set to false. 3000 overrides the default port as well.

This gives you a lot of power to change the options appliciable to each environment.

On the tests in

## .wp-env.override.json

> Any fields here will take precedence over .wp-env.json. This file is useful, when ignored from version control, to persist local development overrides. Note that options like `plugins` and `themes` are not merged. As a result, if you set `plugins` in your override file, this will override all of the plugins listed in the base-level config. The only keys whch are merged are `config` and `mappings`. This means that you can set your own wp-config values without loosing any of the default values.

Reviewer: ```suggestion
Any fields here will take precedence over .wp-env.json. This file is useful when ignored from version control, to persist local development overrides. Note that options like `plugins` and `themes` are not merged. As a result, if you set `plugins` in your override file, this will override all of the plugins listed in the base-level config. The only keys which are merged are `config` and `mappings`. This means that you can set your own wp-config values without losing any of the default values.
```
